Data Management Guidelines (#349)
nf-core/nf-co.re: markdown/usage/data_guidelines.md

---

> title: Data Management Guidelines

Reviewer: How about the title: Resource Estimation and Data sharing.

It's more specific to the aim, and should serve to pull readers in looking to find that information.

Author: ```suggestion
title: Resource Estimation and Data sharing
```

Archiving is often the responsibility of your host institution, contact them for more details.

## Data sharing

In the era of [FAIR](https://www.force11.org/group/fairgroup/fairprinciples) (Findable, Accessible, Interoperable and Reusable) and [Open science](https://www.vr.se/english/mandates/open-science/open-access-to-research-data.html), datasets should be made available to the public, for example by submitting your data to a public repository.

### Choosing a repository

It’s recommended to choose a domain-specific repository when possible. It is also important to consider the sustainability of the repository to ensure that the data will remain public. Please see [SciLifeLab's data guidelines](https://scilifelab-data-guidelines.readthedocs.io/en/latest/docs/index.html) or the [EBI archive wizard](https://www.ebi.ac.uk/submission/) for suggestions depending on data type. You can also refer to the [ELIXIR Deposition Databases](https://elixir-europe.org/services/tag/elixir-deposition-databases) and [Scientific Data’s Recommended Data Repositories](https://www.nature.com/sdata/policies/repositories), to find suitable repositories.

> 

Author: ```suggestion
Also note that some funding agencies have specific requirements regarding data deposition. For example, data generated in projects funded by US federal grants should be deposited into public databases such as [SRA](https://www.ncbi.nlm.nih.gov/sra) for raw sequencing data and [GEO](https://www.ncbi.nlm.nih.gov/geo/) for functional genomic data.
```

> 

Author: ```suggestion
Also note that funding agencies might have specific requirements regarding data depositio. For example, data generated in projects funded by US federal grants should be deposited into public databases such as [SRA](https://www.ncbi.nlm.nih.gov/sra) for raw sequencing data and [GEO](https://www.ncbi.nlm.nih.gov/geo/) for functional genomics data. 
```

---
title: Data Management Guidelines
subtitle:
---

> ## Data management

Reviewer 2: A general introduction here would be nice - I think what @mahesh-panchal suggests is a good idea: spelling out why data management and planning is important, and what value it gives.

Reviewer: What about:

Research councils are recently recognizing the importance of managing data from research. Some now request detailed Data Management Plans (DMP) as part of the grant application. Guidance is provided below to aid the creation of DMP's, estimate resources needed by nf-core workflows, and how to share the resulting data.

Reviewer: Or maybe Research funding agencies instead of research councils.

Author: ```suggestion
## Data management
Funding agencies are recognizing the importance of research data management, and some are now requesting a Data Management Plan (DMP) as part of grant applications. Guidance is provided below to aid the creation of DMPs, estimate resources needed by nf-core pipelines and how to share the resulting data. 
```

Author: ```suggestion
## Data management
Funding agencies are recognizing the importance of research data management and some now request detailed Data Management Plans (DMP) as part of the grant application. Guidance is provided below to aid the creation of DMPs, estimate resources needed by nf-core workflows, and how to share the resulting data.
```

### Data Management Plan

A Data Management Plan (DMP) is a revisable document explaining how you intend to handle new and existing data, during and following the conclusion of your research project. It is wise to write a DMP as early as possible, using either a tool provided by your host institution or [DS Wizard](http://dsw.scilifelab.se/) ([users outside Sweden](https://ds-wizard.org/)). Ethical and legal considerations regarding the data will depend on where the research is conducted, this is especially true for projects including sensitive human data. For more  information about the Swedish context, please review this page on [Sensitive personal data](https://scilifelab-data-guidelines.readthedocs.io/en/latest/docs/general/sensitive_data.html).

### Data storage and computational resources

> To estimate computational resources needed for a specific pipeline see docs/output page ...
> 
> / Feedback from nf-core team needed /: Insert statement/instructions on where or how to get the estimated file sizes, folder structures etc.

Reviewer 2: ```suggestion
To estimate computational resources needed for a specific pipeline please see the pipeline documentation. This lists the different output filetypes you can expect.
In the future we hope to automate a full run of each pipeline after every release. The pipeline docs will then show a full set of results from a real run, along with all file sizes. This can then be used as a guide as to what to expect for your data.
```

### Choosing a repository

It’s recommended to choose a domain-specific repository when possible. It is also important to consider the sustainability of the repository to ensure that the data will remain public. Please see[SciLifeLab's data guidelines](https://scilifelab-data-guidelines.readthedocs.io/en/latest/docs/index.html) or the [EBI archive wizard](https://www.ebi.ac.uk/submission/) for suggestions depending on data type. You can also refer to the [ELIXIR Deposition Databases](https://elixir-europe.org/services/tag/elixir-deposition-databases) and [Scientific Data’s Recommended Data Repositories](https://www.nature.com/sdata/policies/repositories), to find suitable repositories.

For datasets that do not fit into domain-specific repositories, you can use an institutional repository when available (e.g. [SciLifeLab Data Repository](https://scilifelab.figshare.com/)) or a general repository such as [Figshare](https://figshare.com/) and [Zenodo](https://zenodo.org/).

### Preparing for submission

> #### **Describing and organizing your data**

Reviewer 2: ```suggestion
#### Describing and organizing your data
```

> #### **Describing and organizing your data**

Author: ```suggestion
#### Describing and organizing your data
```

Metadata should be provided to help others discover, identify and interpret the data. Researchers are strongly encouraged to use community metadata standards and ontologies where these are in place, consult e.g [FAIRsharing.org](https://fairsharing.org/databases/). Data repositories may also provide guidance about metadata standards and requirements. Capture any additional documentation needed to enable reuse of the data in Readme text files and [Data Dictionaries](https://help.osf.io/hc/en-us/articles/360019739054-How-to-Make-a-Data-Dictionary) that describe what all the variable names and values in your data really mean. Identifiers to refer to e.g. ontology terms can be designed for computers or for people; in a FAIR data context it is recommended to supply both a human-readable as well as a machine-resolvable Persistent Identifier (PID) for each concept used in the data.

> #### **Data integrity**

Reviewer 2: ```suggestion
#### Data integrity
```

Some repositories require md5 checksums to be uploaded along with the files. Also consider adding checks that your data files follow the intended file formats and can be opened by standard software for those formats.

> Consult the docs/output page to see if the pipeline generates the sums automatically and where to find them.
> 
> / Feedback from nf-core team needed /: Or where should this be specified?

Reviewer 2: ```suggestion
If you are using a Linux system, you can generate md5 checksums using the `md5sum` command.
```

I think just omit this for now. Some pipelines already generate md5 sums, some do not. It's perhaps a good idea to make this a guideline for all nf-core pipelines to try to expand this support out, but that should be a separate issue.

> #### **Choosing a license**

Reviewer 2: ```suggestion
#### Choosing a license
```

> To ascertain re-usability data should be released with a clear and accessible data usage license. We suggest making your data available under licences that permit free reuse of data, e.g. a Creative Commons licence, such as CC0 or CC-BY. The [EUDAT licence selector wizard](https://ufal.github.io/public-license-selector/)can help you select suitable licences for your data. Note that sequence data submitted to ENA (or GenBank) are implicitly free to reuse by others as specified in the [INCD Standards and policies]( https://www.ebi.ac.uk/ena/standards-and-policies).

Reviewer 2: ```suggestion
To ascertain re-usability data should be released with a clear and accessible data usage license. We suggest making your data available under licences that permit free reuse of data, e.g. a Creative Commons licence, such as CC0 or CC-BY. The [EUDAT licence selector wizard](https://ufal.github.io/public-license-selector/) can help you select suitable licences for your data. Note that sequence data submitted to ENA (or GenBank) are implicitly free to reuse by others as specified in the [INCD Standards and policies]( https://www.ebi.ac.uk/ena/standards-and-policies).
```

Minor point - something I try to do when writing documentation now is to split lines fairly frequently. It doesn't affect the output formatting at all (on the nf-core website anyway, GitHub does show line breaks), and it means that the diffs are a lot easier to read. All I did here was add a single space after the EUDAT link 😄

> To ascertain re-usability data should be released with a clear and accessible data usage license. We suggest making your data available under licences that permit free reuse of data, e.g. a Creative Commons licence, such as CC0 or CC-BY. The [EUDAT licence selector wizard](https://ufal.github.io/public-license-selector/) can help you select suitable licences for your data. Note that sequence data submitted to ENA (or GenBank) are implicitly free to reuse by others as specified in the [INCD Standards and policies]( https://www.ebi.ac.uk/ena/standards-and-policies).

Author: ```suggestion
To ascertain re-usability data should be released with a clear and accessible data usage license. We suggest making your data available under licences that permit free reuse of data, e.g. a Creative Commons licence, such as CC0 or CC-BY. The [EUDAT licence selector wizard](https://ufal.github.io/public-license-selector/) can help you select suitable licences for your data.
Note that sequence data submitted to [ENA](https://www.ebi.ac.uk/ena)/[GenBank](https://www.ncbi.nlm.nih.gov/genbank/)/[DDBJ](https://www.ddbj.nig.ac.jp/index-e.html) are implicitly free to reuse by others as specified in the [INCD Standards and policies]( https://www.ebi.ac.uk/ena/standards-and-policies).
```
linking to ENA/GenBank/DDBJ

---
title: Resource Estimation and Data sharing

> subtitle:

Reviewer 2: Needs a subtitle 👍

Reviewer 2: Maybe this, but the other way around? The current title is too long really, as it is:

![image](https://user-images.githubusercontent.com/465550/80714248-ed882f80-8af4-11ea-8569-a3477c517bfc.png)

If reversed then it looks better:

![image](https://user-images.githubusercontent.com/465550/80714278-fda00f00-8af4-11ea-8aee-570c4911be2d.png)

Reviewer 2: Maybe something like this instead?

![image](https://user-images.githubusercontent.com/465550/80714362-20cabe80-8af5-11ea-972c-f44019d364eb.png)

```suggestion
subtitle: How to plan your project with resource estimation and data sharing guidelines.
```

Reviewer: I like your suggestion better. My only worry is finding it from the front page.

Reviewer: I really can't think of anything short though which is better.

Author: I think this last suggestion looks good. If this makes the information visible enough for the users or not I can't really tell

> title: Resource Estimation and Data sharing
> subtitle: Data Management Guidelines

Reviewer: ```suggestion
title: Data Management
subtitle: How to plan your project, estimate resources, and share your results.
```

---

## Data management

> Research data management concerns the organization, storage, preservation, and sharing of data that is collected or analyzed during a research project. Proper planning and data management facilitates sharing and allows others to validate and reuse the data. Also, funding agencies are recognizing the importance of research data management and some now request detailed Data Management Plans (DMP) as part of the grant application. Guidance is provided below to aid the creation of DMPs, estimate resources needed by nf-core workflows, and how to share the resulting data.

Reviewer: Would it be ok to start with the motivation, instead of the definition? 

```suggestion
Funding agencies are recognizing the importance of research data management and some now request detailed Data Management Plans (DMP) as part of the grant application. Research data management concerns the organization, storage, preservation, and sharing of data that is collected or analyzed during a research project. Proper planning and data management facilitates sharing and allows others to validate and reuse the data. Guidance is provided below to aid the creation of DMPs, estimate resources needed by nf-core workflows, and how to share the resulting data.
```

(Github has a crazy amount of features. Time to get used to them.)

Author: Sure, starting with the definition sentence might scare some ppl away :)

Backing up and archiving your data is essential. The 3-2-1 rule of thumb means that you should have 3 copies of the data, on 2 different types of media, and 1 of the copies at different physical location. Consider uploading the raw data to a repository already when receiving them, under an embargo (if that is important to you). This way you always have an off-site backup with the added benefit of making the data sharing phase more efficient. Identifying a suitable repository early on will allow you to conform to their standards and metadata requirements already from the start.

Archiving is often the responsibility of your host institution, contact them for more details.

## Data sharing

In the era of [FAIR](https://www.force11.org/group/fairgroup/fairprinciples) (Findable, Accessible, Interoperable and Reusable) and [Open science](https://ec.europa.eu/research/openscience/index.cfm), datasets should be made available to the public, for example by submitting your data to a public repository.

### Choosing a repository

> It’s recommended to choose a domain-specific repository when possible. It is also important to consider the sustainability of the repository to ensure that the data will remain public. Please see [SciLifeLab's data guidelines](https://scilifelab-data-guidelines.readthedocs.io/en/latest/docs/index.html) or the [EBI archive wizard](https://www.ebi.ac.uk/submission/) for suggestions depending on data type. You can also refer to the [ELIXIR Deposition Databases](https://elixir-europe.org/services/tag/elixir-deposition-databases), [Scientific Data’s Recommended Data Repositories](https://www.nature.com/sdata/policies/repositories), [FAIRsharing.org](https://fairsharing.org/databases/) and [re3data.org](https://www.re3data.org/) to find suitable repositories.

Reviewer 3: I would suggest flipping the first Scilifelab and and ENA links, as the latter is more applicable across other countries/institutes

Author: ```suggestion
It’s recommended to choose a domain-specific repository when possible. It is also important to consider the sustainability of the repository to ensure that the data will remain public. Please see the [EBI archive wizard](https://www.ebi.ac.uk/submission/) or [SciLifeLab's data guidelines](https://scilifelab-data-guidelines.readthedocs.io/en/latest/docs/index.html) for suggestions depending on data type. You can also refer to the [ELIXIR Deposition Databases](https://elixir-europe.org/services/tag/elixir-deposition-databases), [Scientific Data’s Recommended Data Repositories](https://www.nature.com/sdata/policies/repositories), [FAIRsharing.org](https://fairsharing.org/databases/) and [re3data.org](https://www.re3data.org/) to find suitable repositories.
```
